nrf_modem/Kconfig

```diff
@@ -1,18 +1,27 @@
+# Selected by NRF_MODEM_LIB in sdk-nrf
 config NRF_MODEM
 	bool
 	imply FPU_SHARING
 
+menu "nrf_modem (Modem library)"
```

Comment:
I wondered the same when looking at it. I think it is because the is a Kconfig for the integration layer (`NRF_MODEM_LIB`) which is what pulls everything in already (and `select NRF_MODEM`).